l1-contracts/src/core/libraries/crypto/EIP712Lib.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Aztec Labs.
+pragma solidity >=0.8.27;
+
+library EIP712Lib {
+  struct EIP712Domain {
+    string name;
+    string version;
+  }
+
+  bytes32 public constant EIP712DOMAIN_TYPEHASH =
+    keccak256("EIP712Domain(string name,string version)");
+
+  bytes32 public constant DOMAIN_SEPARATOR =
```

Comment:
Yes, I was thinking we could have it this way until the deployed address was certain, which would allow clients to create signatures without needing to query the contract to get the requisite info. I'm fine to remove in favor of inheriting openzeppelin's EIP712 as mentioned.